source/image/ImageSet.cpp

```diff
@@ -146,18 +145,24 @@ namespace {
 				+ (ignored > 1 ? " frames" : " frame") + " ignored in total).");
 		}
 	}
+
+	string ToLowerCase(std::string str)
+	{
+		std::transform(str.begin(), str.end(), str.begin(), [](unsigned char c){ return std::tolower(c); });
+		return str;
+	}
 }
 
 
 
 // Check if the given path is to an image of a valid file type.
 bool ImageSet::IsImage(const string &path)
 {
-	if(path.length() < 4)
+	if(path.find('.') == string::npos)
 		return false;
 
-	string ext = path.substr(path.length() - 4);
-	return (ext == ".png" || ext == ".jpg" || ext == ".PNG" || ext == ".JPG");
+	string ext = path.substr(path.find_last_of('.') + 1);
```

Comment:
Just calculate `path.find_last_of('.')` at the beginning and also use it for the early return, it will also return `string::npos` if no matches are found.